src/mirrors_config.vala

```diff
@@ -116,16 +116,20 @@ namespace Pamac {
 						unowned Variant variant;
 						if (line.contains ("Method")) {
 							if (new_conf.lookup_extended ("Method", null, out variant)) {
-								data.append ("Method=%s\n".printf (variant.get_string ()));
+								if (variant.get_string () == "rank") {
+									data.append ("# Method = \n".printf (line));
+								} else {
+									data.append ("Method = %s\n".printf (variant.get_string ()));
+								}
 							} else {
 								data.append (line + "\n");
 							}
 						} else if (line.contains ("OnlyCountry")) {
 							if (new_conf.lookup_extended ("OnlyCountry", null, out variant)) {
 								if (variant.get_string () == "ALL") {
-									data.append ("#%s\n".printf (line));
+									data.append ("# OnlyCountry = \n".printf (line));
```

Comment:
Sorry about that. Though it seems very weird to me.

    data.append ("OnlyCountry = %s\n".printf (variant.get_string ()));

This line will fail as well?

Actually I copied the second if clause from OnlyCountry to Method and and changed OnlyCountry to Method and added spaces. I didn't think about testing it - it seemed totally logic to me - copying a working piece of code.

```
								if (variant.get_string () == "ALL") {
									data.append ("# OnlyCountry = \n".printf (line));
								} else {
									data.append ("OnlyCountry = %s\n".printf (variant.get_string ()));								}
```